src/com/sap/piper/MapUtils.groovy

```diff
@@ -46,18 +41,16 @@ class MapUtils implements Serializable {
      *        in <code>m</code> in a recursive manner.
      * @param strategy Strategy applied to all non-map entries
      */
-    @NonCPS
     static void traverse(Map m, Closure strategy) {
 
         def updates = [:]
-        for(def e : m.entrySet()) {
-            if(isMap(e.value)) {
-                traverse(e.getValue(), strategy)
-            }
-            else {
+        m.each { key, value ->
```

Comment:
is this change cps related?

Reply:
We refactored a lot of code around `.entrySet()` calls to avoid issues with `Iterator`s.